### ISSUE TITLE
Using 2 esp8266 to avoid overlapping channels

### ISSUE THREAD

**Issue by butz6617** [martin-ger/esp_wifi_repeater#78]
First I made the Repeater with a esp 01, worked good. But the Range wasnt that good as i hoped.
So i ordered a couple of esp 02 with antennas. The Rabge is way better, but the Problem that both the SoftAP and the STA have the same channel seemed to get bigger as the esp 02 with seems really strong. I know this was already a issue and has been closed, but i have a Approach which may help us. Using two esp8266, as they are fairly cheap 1$ to 2$, why not one is connected to Sta and one makes the AP. As im not really into esp8266 that long, i only know it has a (H)SPI and UART, but i dont kniw if SPI can be used for the communication between 2 esps, i guess not because Master Slave System and i dont know if the Uart is fast enough to keep up with the 5 mbit (i even got 8 mbit). Someone ince mentioned something about 40x115200 baud is the Maximum. What do you think about this approach, or would the Benefit not be that big. I noticed that Repeaters which you can buy, repeat on the same channel as the STA. So its not that big of a Problem?

**Comment by martin-ger** [comment]
Of course, repeating on the same channel rquires twice the bandwith of the original signal, that's why some Wireless Distribution System (APs with WLAN backbone) use a mix of 2,4 GHz and 5 GHz channels for backbone and AP. Using two different channels is in in general an option to increase bandwith.

But I don't think that this can will work here. I did some experiments with the UART with my SLIP-router: the UART has a small buffer of 128 byte, no hw-flow control, and reading (and writing) it requires good timing. I doubt it will work reliabliy at several MHz speed. I have no experience with SPI either, but as far as I know nobody has managed to connect two EPSs via SPI.

**Comment by hackrid** [comment]
Connecting 2 esps with spi should be perfectly simple. But it would requiere a lwip netif that read and writes from and to spi. Would this. Be complicated? 
This might be a good idea input: http://lwip.wikia.com/wiki/Writing_a_device_driver

I have a different application in mind where the second esp opens a "low rate " network to extend the range even more

**Comment by martin-ger** [comment]
Driver is not the issue - look at the SLIP driver here: https://github.com/martin-ger/esp_slip_router .
I would need code for the SPI communication.

**Comment by hackrid** [comment]
I think that's manageable☺️

Did you check the spi examples in idf? 

_Sent from my Lge LG-D855 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
